### ISSUE TITLE
feat(schematics/lib): add directory option for lib and app schematics

### ISSUE THREAD

**Issue by jschwarty** [nrwl/nx#86]
Add an option to the `lib` and `app` schematics to provide the directory where it will be created. Example:
```
ng g lib myLib --directory=libs/helpers

ng g lib generalUI --directory=libs/shared-components
```

This would allow users to create nested folder structures for their libs and apps.

Schematic would need to use the `directory` value for the file locations and use it in conjunction with the `sourceDir` when setting the `root` property of the app in the `.angular-cli.json` file.

**Comment by zpydee** [comment]
i would personally prefer `ng g lib nested-dir-name/lib-name` - less verbose and imho, more expressive

**Comment by zpydee** [comment]
also, need capability in `tslint.json` to specify nested directory names that are allowed

**Comment by zpydee** [comment]
is it possibly necessary to support multi-level nesting? i.e.

```
ng g lib demo/features
ng g lib demo/features/home
ng g lib demo/features/widgets
```

**Comment by zpydee** [comment]
i am currently reorganising my workspace into a more granular pattern in anticipation of you supporting nested libraries, and some feedback on your thoughts of how you expect to approach this would be nice (please). For example, will you support multiple nesting layers?

**Comment by vsavkin** [comment]
Fixed here 013a828d1e31f55a9b3f7c69587316890ea834d4

**Comment by beeman** [comment]
Awesome @vsavkin! Looking forward to trying it out! 👍🏼
